packages/subproviders/src/subproviders/ledger.ts

```diff
@@ -211,6 +183,7 @@ export class LedgerSubprovider extends BaseWalletSubprovider {
         }
         const ledgerEthereumClient = await this._ledgerEthereumClientFactoryAsync();
         this._connectionLock.release();
+        this._ledgerClientIfExists = ledgerEthereumClient;
```

Comment:
Why was this added? Was this missing, i.e a bug?

Reply:
It could've been a bug if a developer asked for a lock and forgot to assign it or assigned it to a local variable. I moved it in here to make sure this doesn't happen in the future.

Reply:
The problem with only assigning it in that method is that ts forces us to check if it is undefined, even when that method throws.

```typescript
await this._createLedgerClientAsync();
await this._createLedgerClientAsync();
        if (_.isUndefined(this._ledgerClientIfExists)) {
            throw new Error('aaa');
        }

        let ledgerResponse;
        try {
            ledgerResponse = await this._ledgerClientIfExists.getAddress(
...
```
versus
```typescript
        const ledgerClient = await this._createLedgerClientAsync();

        let ledgerResponse;
        try {
            ledgerResponse = await ledgerClient.getAddress(
```

Thoughts?

Reply:
Ok, then let's remove this assignment. It's not clean to have both. I do think this is a candidate for a future refactor.